src/vs/workbench/services/untitled/common/untitledTextEditorModel.ts

```diff
@@ -141,6 +143,8 @@ export class UntitledTextEditorModel extends BaseTextEditorModel implements IUnt
 			this.setModeInternal(preferredMode);
 		}
 
+		this._autoDetectLanguageScheduler = this._register(new RunOnceScheduler(() => this.autoDetectLanguage(), 600));
```

Comment:
Even more concise, you can move this entire line to the top where you declare the member.